model/task/task.go

```diff
@@ -1264,6 +1252,12 @@ func SetTasksScheduledTime(tasks []Task, scheduledTime time.Time) error {
 	return nil
 }
 
+// GetTaskIdBetweenIds retrieves the task id between two tasks that are of the same.
+func GetTaskIdBetweenTasks(t1, t2 Task) (*Task, error) {
```

Comment:
Are we assuming that `BuildVariant`, `DisplayName`, `Project`, and `Requester` are all the same between `t1` and `t2`, or can they differ? If so, can we validate that?